providers/SCConfigurationProvider.js

```diff
@@ -2,7 +2,17 @@
 
     'use strict';
 
-    angular.module('soundcloudify.core').provider('SCConfiguration', SCConfigurationProvider);
+    try{
+        angular.module('soundcloudify.core')
+            .provider('SCConfiguration', SCConfigurationProvider);
+    }
+    catch (err){ /*ignore*/ }
+
+    try{
+        angular.module('soundcloudify.background')
+            .provider('SCConfiguration', SCConfigurationProvider);
+    }
```

Comment:
I can't inject soundcloudify.core as a dependency because the applications
are on two different pages. The background app is in the background page
and soundcloudify.core is in the popup page. You can't inject from one page
to another.

On Fri, Aug 14, 2015 at 10:25 PM, Anh Trinh <notifications@github.com>
wrote:

> In providers/SCConfigurationProvider.js
> <https://github.com/ptgamr/soundcloudify-client-core/pull/2#discussion_r37132751>
> :
>
> > @@ -2,7 +2,17 @@
> >
> >      'use strict';
> >
> > -    angular.module('soundcloudify.core').provider('SCConfiguration', SCConfigurationProvider);
> > +    try{
> > +        angular.module('soundcloudify.core')
> > +            .provider('SCConfiguration', SCConfigurationProvider);
> > +    }
> > +    catch (err){ /*ignore*/ }
> > +
> > +    try{
> > +        angular.module('soundcloudify.background')
> > +            .provider('SCConfiguration', SCConfigurationProvider);
> > +    }
>
> Well, sorry that I didn't replied you earlier via email. I think this try
> catch approach isn't the way angular module is designed for.
>
> These files located in core should always be in core. If you want to use
> the service in the background app, you just need to inject the
> soundcloudify.core module into the background app.
>
> Please refer to this comment:
> https://github.com/ptgamr/upnext/pull/29/files#diff-2d77636971b1859b946f98b997c87b3cR22
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ptgamr/soundcloudify-client-core/pull/2/files#r37132751>
> .
>



-- 
*Duncan Kelly*
Cloud Software Engineer
Perceptive Software | from Lexmark

(O)  +1 913 307 2238 | (M)  +1 913 808 1376
www.perceptivesoftware.com

<http://inspire.perceptivesoftware.com/> <http://www.incontextmag.com/>
<https://twitter.com/perceptivesw>
<https://www.facebook.com/perceptivesw>
<http://www.linkedin.com/company/10715?trk=tyah>